src/core/server.js

```diff
@@ -800,7 +810,16 @@ Server.prototype._sendMessage = function(message) {
     if (this._remote.trace) {
       log.info(this.getServerID(), 'request:', message);
     }
-    this._ws.send(JSON.stringify(message));
+    this._ws.send(JSON.stringify(message), (error) => {
+      // sometimes gives 'not opened'
+      // without callback it wil throw
+      if (error) {
+        // resend in case of error
+        this.once('connect', () => {
```

Comment:
This worries me a bit. If there is any session-based message, it will probably produce unexpected behavior. Like if you try to close a pathfinding stream, but the message gets sent after a reconnect, the server might reply "there is no stream to close", which would be confusing if you didn't see any errors. This example is fairly innocuous, but I can imagine more problematic situations arising. But I guess this issue is baked into the whole system, so I suppose we can leave this for now.